src/modules/user/Login.jsx

```diff
@@ -1,69 +1,188 @@
 import React, { useState } from "react";
 import { useUserDispatch } from "./UserContext";
 import { useNavigate } from "react-router-dom";
-import { Container } from "../../shared-component/Container/index";
-import { Button } from "../../shared-component/Button/index";
-import { Input } from "../../shared-component/Input/index";
 import { validateEmail } from "../../utils";
+import Container from "@mui/material/Container";
+import {
+  Avatar,
+  Box,
+  Button,
+  Paper,
+  Stack,
+  TextField,
+  Typography,
+  FormControl,
+  FormControlLabel,
+  Checkbox,
+  InputAdornment,
+} from "@mui/material";
+import LockOutlinedIcon from "@mui/icons-material/LockOutlined";
+import IconButton from "@mui/material/IconButton";
+import Visibility from "@mui/icons-material/Visibility";
+import VisibilityOff from "@mui/icons-material/VisibilityOff";
 
 export const Login = () => {
   const dispatch = useUserDispatch();
   const [username, setUsername] = useState("");
   const [password, setPassword] = useState("");
   const [error, setError] = useState("");
+  const [userNameError, setUsernameError] = useState("");
+  const [passwordError, setpasswordError] = useState("");
+
   const navigate = useNavigate();
 
   const handleLogin = () => {
+    let hasError = false;
+
     if (!username || !password) {
       setError("Username and password are required.");
-      return;
-    }
-    if (!validateEmail(username)) {
-      setError("Invalid email format");
-      return;
+      hasError = true;
+    } else if (!validateEmail(username)) {
+      setUsernameError("Invalid email format");
+      hasError = true;
+    } else {
+      setUsernameError("");
     }
     if (password.length < 4) {
-      setError("Password must be at least 4 characters.");
-      return;
+      setpasswordError("Password must be at least 4 characters.");
+      hasError = true;
+    } else {
+      setpasswordError("");
     }
+    if (hasError) return;
 
     dispatch({ type: "LOGIN", payload: { username, password } });
```

Comment:
From what I see, you don't have LOGIN action in the UserContext. You need to add it to be able to use it.